### PR TITLE
Refactor Pets table to add weight attributes

### PR DESCRIPTION
**Pull request by kasugaijin**
### Describe your change in plain English.
Remove `size` column from Pets table and create a new `weights` table that allows a pet to have a `from` weight and a `to` weight and a unit of `lb` or `kg`. This was a requested feature by Baja Pet Rescue to provide more flexibility on weight ranges.

### Link to the issue
[Issue 14](https://github.com/rubyforgood/pet-rescue/issues/14)

incidental UI change on date picker for age to make styling fit in with rest of pet form
image

### CODE REVIEW

**Review by edwinthinks — CHANGES_REQUESTED** [review]
Hey there! Thanks @kasugaijin for kicking off a good discussion amongst the team with this PR. I added a few change requests but also some just suggestions to talk about (non-blocking items)

Looking forward to seeing where we go with this work! 🚀

File: app/models/pet.rb [hunk 1]
```diff
@@ -48,6 +53,20 @@ class Pet < ApplicationRecord
     :opening_soon,
     :paused_until_further_notice]
 
+  def self.weight_units
+    ["lb", "kg"]
+  end
```
[the remark this answers is not in the document]
Comment (kasugaijin) [reply app/models/pet.rb:58]: Good point. Done!

File: app/models/pet.rb [hunk 2]
```diff
@@ -32,9 +34,12 @@ class Pet < ApplicationRecord
 
   validates :name, presence: true
   validates :birth_date, presence: true
-  validates :size, presence: true
   validates :breed, presence: true
   validates :sex, presence: true
+  validates :weight_from, presence: true
+  validates :weight_to, presence: true
+  validates :weight_unit, presence: true
+  validate :to_weight_must_be_greater_than_from_weight
```
[the remark this answers is not in the document]
Comment (kasugaijin) [reply app/models/pet.rb:42]: yes this is much cleaner. Done.

File: app/models/weight.rb [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Indexes
+#
+#  index_weights_on_pet_id  (pet_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (pet_id => pets.id)
+#
+class Weight < ApplicationRecord
```
[the remark this answers is not in the document]
Comment (kasugaijin) [reply app/models/weight.rb:21]: Woops. Done.

File: app/views/organization_pets/_form.html.erb [hunk 4]
```diff
@@ -48,20 +50,33 @@
...
+            <%= form.label :weight_from, 'From weight' %>
+            <%= form.text_field :weight_from, class: 'form-control' %>
+            <% pet.errors.full_messages_for(:weight_from).each do |error| %>
+              <div class="alert alert-danger mt-1" role="alert"><%= error.message %></div>
+            <% end %>
+          </div>
+          <div class='me-3'>
+            <%= form.label :weight_to, 'To weight' %>
+            <%= form.text_field :weight_to, class: 'form-control' %>
+            <% pet.custom_messages(:weight_to).each do |error| %>
```
[the remark this answers is not in the document]
Comment (kasugaijin) [reply app/views/organization_pets/_form.html.erb:65]: Done.

File: app/models/pet.rb [hunk 5]
```diff
@@ -48,6 +53,20 @@ class Pet < ApplicationRecord
...
+
+  def to_weight_must_be_greater_than_from_weight
+    if weight_from.present? && weight_to.present?
+      errors.add(:weight_to, "Must be greater than from weight") if weight_to < weight_from
+    end
+  end
+
+  def custom_messages(attribute)
+    errors.where(attribute)
+  end
```
[the remark this answers is not in the document]
Comment (kasugaijin) [reply app/models/pet.rb:66]: Yes using the ComparisonValidator I can do away with the custom validation and error message.
I agree we need to add validation error messages to translation filesm though I think that could be a separate PR to do them all at once. That and I can get this one in as it's been a lot of refactoring already :)

[the remark this answers is not in the document]
Comment (edwinthinks) [reply app/models/pet.rb:66]: Sounds good to me, could you create the issue?

File: app/models/weight.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (edwinthinks) [reply app/models/weight.rb:21]: Looks like we still have this file that we intended to remove?

File: app/models/pet.rb [hunk 6]
```diff
@@ -32,9 +34,12 @@ class Pet < ApplicationRecord
 
   validates :name, presence: true
   validates :birth_date, presence: true
-  validates :size, presence: true
   validates :breed, presence: true
   validates :sex, presence: true
+  validates :weight_from, presence: true
+  validates :weight_to, presence: true
+  validates :weight_unit, presence: true
```
Comment (edwinthinks) [inline app/models/pet.rb:41]: Let's add the inclusion to make sure it only is the two values we expect - https://guides.rubyonrails.org/active_record_validations.html#inclusion

File: app/models/pet.rb [hunk 7]
```diff
@@ -32,9 +34,12 @@ class Pet < ApplicationRecord
 
   validates :name, presence: true
   validates :birth_date, presence: true
-  validates :size, presence: true
   validates :breed, presence: true
   validates :sex, presence: true
+  validates :weight_from, presence: true
+  validates :weight_to, presence: true
```
Comment (edwinthinks) [inline app/models/pet.rb:40]: Let's force them to be numerical in case of accidentally adding a string or none number item - https://guides.rubyonrails.org/active_record_validations.html#numericality

Like must be greater than 0 for example

File: test/models/pet_test.rb [hunk 8]
```diff
@@ -12,11 +12,13 @@ class PetTest < ActiveSupport::TestCase
...
     should validate_presence_of(:name)
     should validate_presence_of(:birth_date)
-    should validate_presence_of(:size)
     should validate_presence_of(:breed)
     should validate_presence_of(:sex)
     should validate_presence_of(:description)
     should validate_length_of(:description).is_at_most(1000)
+    should validate_presence_of(:weight_from)
+    should validate_presence_of(:weight_to)
+    should validate_presence_of(:weight_unit)
```
Comment (edwinthinks) [inline test/models/pet_test.rb:21]: Once you've updated those tests in my comment above you can add them here.

Here is the documentation of the library https://github.com/thoughtbot/shoulda-matchers with associated matchers.

**Review by edwinthinks — COMMENTED** [review]
Looking great! A few things is I think there is some files still checked in + and improving the validations a bit on the model. Once those are set, I think this is good to go!

File: app/models/weight.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (kasugaijin) [reply app/models/weight.rb:21]: Confirmed deleted this time. I am not sure what happened there.

File: app/models/pet.rb (hunk 6 above)
Comment (kasugaijin) [reply app/models/pet.rb:41]: Good call. Done.

File: app/models/pet.rb (hunk 7 above)
Comment (kasugaijin) [reply app/models/pet.rb:40]: Yes good call. Done.

**Review by edwinthinks — APPROVED** [review]
Boom! LGTM! Very nice!
